src/app/shared/services/applications-service.ts

```diff
@@ -30,6 +31,10 @@ export class ApplicationsService {
         return requests.get(`/applications/${name}/manifests`).query({name, revision}).then((res) => res.body as models.ManifestResponse);
     }
 
+    public diffManifests(appName: string, resourceName?: string, APIVersion?: string, kind?: string): Promise<models.ManifestDiffResponse> {
+        return requests.post(`/applications/${appName}/diff`).send({resourceName, APIVersion, kind}).then((res) => res.body as models.ManifestDiffResponse);
```

Comment:
API returns list of strings not ManifestDiffResponse . Please use `JSON.parse` to deserialize strings into proper response.